git-record/src/ui.rs

```diff
@@ -0,0 +1,1041 @@
+                        global_changed_hunk_num += 1;
+                        let description = format!(
+                            "hunk {}/{} in current file, {}/{} total",
+                            hunk_num,
+                            local_num_changed_hunks,
+                            global_changed_hunk_num,
+                            global_num_changed_hunks
+                        );
+
+                        self.make_changed_hunk_views(
+                            main_tx.clone(),
+                            &mut file_view,
+                            &mut line_num,
+                            file_num,
+                            hunk_num,
+                            description,
+                            before,
+                            after,
+                        );
+                    }
+                }
+            }
+
+            view.add_child(
+                LinearLayout::horizontal()
+                    .child(
+                        TristateBox::new()
+                            .with_state({
+                                match all_are_same_value(iter_file_changed_lines(file_hunks).map(
+                                    |(
+                                        _hunk_type,
+                                        HunkChangedLine {
+                                            is_selected,
+                                            line: _,
+                                        },
+                                    )| is_selected,
+                                )) {
+                                    SameValueResult::Empty | SameValueResult::AllSame(true) => {
+                                        Tristate::Checked
+                                    }
+                                    SameValueResult::AllSame(false) => Tristate::Unchecked,
+                                    SameValueResult::SomeDifferent => Tristate::Partial,
+                                }
+                            })
+                            .on_change({
+                                let main_tx = main_tx.clone();
+                                move |_, new_value| {
+                                    let file_key = FileKey { file_num };
+                                    if main_tx
+                                        .send(Message::ToggleFile(file_key, new_value))
+                                        .is_err()
+                                    {
+                                        // Do nothing.
+                                    }
+                                }
+                            })
+                            .with_name(file_key.to_string()),
+                    )
+                    .child(TextView::new({
+                        let mut s = StyledString::new();
+                        s.append_plain(" /");
+                        s.append_styled(path.to_string_lossy(), Effect::Bold);
```

Comment:
I wonder if the path should be a `String` to start with. Does this crate need it to be a `PathBuf`?